scripts/API.py

```diff
@@ -0,0 +1,438 @@
+
+	# Check if previous command succeeded
+	if currentBranch is None:
+		return False
+
+	# Update branch
+	retCode = runInsistentJob("git fetch")[0]
+
+	# Check if command succeeded
+	if retCode != 0:
+		return False
+
+	# Get latest local commit
+	localCommit = runJob(f"git rev-parse {currentBranch}")[1]
+
+	# Get latest remote commit
+	retCode, remoteCommit = runInsistentJob(
+		f"git rev-parse origin/{currentBranch}")
+
+	# Check if command succeeded
+	if retCode != 0:
+		return False
+
+	# Return commit comparison
+	return localCommit == remoteCommit
+
+
+def runInsistentJob(cmd: str, cwd: str = './', showOutput: bool = False,
+                    showError: bool = False, showCommand: bool = False,
+                    nRetries: int = 5) -> Tuple[int, str]:
+	"""
+	### This function runs the given network command and retries it the number given of times until one of the succeeds or it fails for all the retries.
+
+	#### Params:
+	- cmd (str): Command to run.
+	- cwd (str): Optional. Path to run the command from. Default is current directory.
+	- showOutput (bool): Optional. If True, output is printed.
+	- showError (bool): Optional. If True, errors are printed.
+	- showCommand (bool): Optional. If True, command is printed in blue.
+	- nRetries (int): Optional. Maximum number of retries for the command.
+
+	#### Returns:
+	- (int): Return code.
+	- (str): Output of the command, regardless of if it is an error or regular output.
+	"""
+	# Running command up to nRetries times (improves resistance to small network errors)
+	for _ in range(nRetries):
+		output=[]
+		retCode = runJob(cmd, cwd=cwd, log=output, show_output=False,show_command=False)
+		# Break loop if success was achieved
+		if retCode == 0:
+			break
+	if len(output) > 0:
+		output = output[0]
+	# Enforce message showing deppending on value
+	if showCommand:
+		print(cmd)
+	if showOutput:
+		print('{}\n'.format(output))
+	if showError:
+		print(output)
+
+	# Returning output and return code
+	return retCode, output
+
+
+
+
+'''
+curl --header "Content-Type: application/json" -X POST --data '{"user": {"userId": "curl attempt"}, "version": {"os": "Ubuntu 22.04.3 LTS", "architecture": "skylake", "cuda": "11.4", "cmake": "null", "gcc": "10.5.0", "gpp": "10.5.0", "scons": "null"}, "xmipp": {"branch": "Unknow", "updated": false}, "returnCode": 0, "logTail": "null"}' --request POST xmipp.i2pc.es/api/attempts/
+
+
+curl --header "Content-Type: application/json" -X POST --data '{"user": {"userId": "f0ccfe1cac91db754d039cf3bb5e7f46327ef3c5442d245b4c5e3b28086003c5"}, "version": {"os": "Ubuntu 22.04.3 LTS", "architecture": "skylake", "cuda": "11.4", "cmake": "null", "gcc": "10.5.0", "gpp": "10.5.0", "mpi": null, "python": null, "sqlite": null, "java": null, "hdf5": null, "jpeg": null}, "xmipp": {"branch": "Unknow", "updated": false}, "returnCode": 0, "logTail": null}' --request POST xmipp.i2pc.es/api/attempts/
+
+'''
+
+if __name__ == '__main__':
+	sendApiPOST(retCode=0)
```

Comment:
Also this, it doesn't make much sense if it's not as a temporary test method.

The reason is that it will always assume `retCode=0` A.K.A. it all went well (even if it didn't).

I can imagine the reason for pushing this api client quickly into the upcoming release is to be able to gather installation reports during the time it takes up to the next release, but I can't see how useful can that data be if it's going to be mostly placeholders (for the fields this new installer cannot fill), and the installation results always fake. And i guess this script will have to be called from command line before running the actual installation script, or from within the first lines of the script (doesn't look like it's implemented since this is the only file modified in the PR), otherwise if it fails, this will never get executed.

The only pro for using the API so soon is to gather data about the versions of some components of user's systems in advance. Would just that be worth the effort + potential issues it can cause?

Reply:
It was there just to debug the code! and I forgot to remove it.